plasmapy/atomic/ionization_states.py

```diff
@@ -241,20 +239,45 @@ def __eq__(self, other):
         if self.base_particles != other.base_particles:
             raise AtomicError
 
-        tol = np.min([self.tol, other.tol])
+        min_tol = np.min([self.tol, other.tol])
 
-        for element in self.base_particles:
+        # Check any of a whole bunch of equality measures, recalling
+        # that np.nan == np.nan is False.
 
-            are_all_close = np.allclose(
-                self.ionic_fractions[element],
-                other.ionic_fractions[element],
-                atol=tol,
-                rtol=0,
-            )
+        for attribute in ['T_e', 'n_e', 'kappa']:
+            this = eval(f"self.{attribute}")
+            that = eval(f"other.{attribute}")
+
+            this_equals_that = np.any([
+                this == that,
+                this is that,
+                np.isnan(this) and np.isnan(that),
+                np.isinf(this) and np.isinf(that),
+                u.quantity.allclose(this, that, rtol=min_tol),
+            ])
```

Comment:
Yeah, it's a pain to arbitrarily compare things that might be `nan`, `inf`, a `Quantity`, and perhaps even things that don't have `__eq__` implemented.  Might be worth creating a function called something like `same_enough` in `plasmapy.utils.pytest_helpers.py` that holds all of this, especially due to code duplication between `IonizationState` and `IonizationStates`.  Other suggestions on how to do this better would be appreciated!

I'm kind of proud of the names `this` and `that`, since it makes the code really readable (to me at least).

Reply:
That it does, it gave me a wide smile when I saw that :D